src/people.h

```diff
@@ -133,11 +71,8 @@ class Guests {
 	};
 	Complaint complaints[COMPLAINT_COUNT];  ///< Statistics about all complaint types.
 
-	bool FindNextFreeGuest();
-	bool FindNextFreeGuest() const;
-	bool HasFreeGuests() const;
-	void AddFree(Guest *g);
-	Guest *GetFree();
+	std::vector<std::unique_ptr<Guest[]>> guests;  ///< All guest slots.
+	std::vector<int> free_guest_indices;           ///< Unused indices in %guests.
```

Comment:
I wondered about pre-allocation of `std::vector`, but the `unique_ptr<>` will avoid that problem. Nice!

A bitset is an alternative to store a collection of integers (although for 1000 free guests, you're using 4000 bytes here, not something to worry about :) )  
While reading a bit about `std::bitset` and it's dynamic cousin `std::vector<bool>`, I am missing the "get index of next set bit" function like the Java version has: https://docs.oracle.com/javase/7/docs/api/java/util/BitSet.html#nextSetBit(int)  included since 2002.  
Without that function, the bitset is next to useless for this purpose, so unless you want to write a bitset implementation, you likely already have the better solution :)